include/bsp/xi_bsp_io_fs.h

```diff
@@ -43,34 +61,127 @@ typedef struct xi_fs_stat_s
 
 /**
  * @enum xi_fs_open_flags_t
+ * @brief Abstracted values that represent the various types of file operations
+ * that should be passed to the underlying system when opening a file resource
+ * via xi_bsp_io_fs_open().
+
+ * As a bitmask there could be more than one of these flags set on a given
+ * open request.
  */
 typedef enum {
     XI_FS_OPEN_READ   = 1 << 0,
     XI_FS_OPEN_WRITE  = 1 << 1,
     XI_FS_OPEN_APPEND = 1 << 2,
 } xi_fs_open_flags_t;
 
+
+/**
+ * @function
+ * @brief Used by the Xively C Client to determine the existance and size of a
+ * file.
+ *
+ * @param [in] resource_name the name of the file to check
+ * @param [out] resource_stat a structure to be populated based on the file
+ * size data.
+ *
+ * @return xi_state_t XI_STATE_OK in case of a success and other in case of an
+ * error.
+ */
 xi_state_t
 xi_bsp_io_fs_stat( const char* const resource_name, xi_fs_stat_t* resource_stat );
 
+/**
+ * @function
+ * @brief Requests that the file be opened for reading or writing, and that
+ * a handle to the file be returned via an out parameter.
+ *
+ * @param [in] resource_name the name of the file to open
+ * @param [in] size the size of file in bytes, useful when opening
+ * files for write.
+ * @param [in] open_flags a read/write/append bitmask of operations as
+ * defined by a bitmask type xi_fs_open_flags_t.
+ * @param [out] resource_handle_out a pointer to an abstracted
+ * xi_fs_resource_handle_t data type. This value will be passed to
+ * future file operations such as read, write or close.
+ *
+ * @return xi_state_t XI_STATE_OK in case of a success and other in case of an
+ * error.
+ */
 xi_state_t xi_bsp_io_fs_open( const char* const resource_name,
                               const size_t size,
                               const xi_fs_open_flags_t open_flags,
                               xi_fs_resource_handle_t* resource_handle_out );
 
+/**
+ * @function
+ * @brief reads a subset of a file that was previously opened
+ * with a call to xi_bsp_io_fs_open.
+ *
+ * @param [in] resource_handle the handle created by a previous call
+ * to xi_bsp_io_fs_open.
+ * @param [in] offset the position of the resource, in bytes, from which
+ * to start read operations.
+ * @param [out] buffer a pointer to a byte array. Data read from the
```

Comment:
I'd say rather this: "An outgoing pointer on a buffer containing the read bytes. The BSP implementation is responsible for all memory management of the buffer, including: allocation, deallocation."
If you think the "starting at offset zero" is informative, then pls. add it somewhere, thx.